chain/src/chain.rs

```diff
@@ -634,9 +636,10 @@ impl Chain {
 		txhashset.merkle_proof(commit)
 	}
 
-	/// Returns current txhashset roots.
-	pub fn get_txhashset_roots(&self) -> TxHashSetRoots {
-		self.txhashset.read().roots()
+	/// Returns current txhashset roots along with current head header.
+	pub fn get_txhashset_roots(&self) -> Result<(TxHashSetRoots, BlockHeader), Error> {
```

Comment:
shouldn't this change name to get_txhashset_roots_and_header ?

Reply:
Actually realised this can be cleaned up - its only used by the api in one place and there is an easier way of getting the "txhashset" roots. We just need the header itself, no need to recalculate the roots off the actual PMMRs.